lib/rules/no-private-routing-service.js

```diff
@@ -93,6 +104,33 @@ module.exports = {
           context.report({ node, message: ROUTER_MICROLIB_ERROR_MESSAGE });
         }
       },
+
+      CallExpression(node) {
+        checkForRouterMain(node);
+      },
+
+      OptionalCallExpression(node) {
+        checkForRouterMain(node);
+      },
     };
+
+    function checkForRouterMain(node) {
+      // Looks for expressions like these:
+      // x.lookup('router:main')
+      // x.y.lookup('router:main')
+      // x?.lookup('router:main')
+      if (
+        catchRouterMain &&
+        (types.isMemberExpression(node.callee) || types.isOptionalMemberExpression(node.callee)) &&
```

Comment:
It could make sense to make an `isOptionalMemberOrMemberExpression` method.

Reply:
That makes sense given we'll probably need to use this combination in many places...

Reply:
It could almost make sense to rename `isMemberExpression` to `isNonoptionalMemberExpression`, to remind people that they should also consider the optional version. Though that could stray to far from the simple `is...` methods.

Reply:
I like that idea.